### PR TITLE
[ITG-45] Przekierowanie do szczegółów zadania po kliknięciu w zadanie w relacji

### PR DESCRIPTION
**Pull request by koszar91**

### CODE REVIEW

File: cabra/src/pages/components/RelatedTask.tsx [hunk 1]
```diff
@@ -49,7 +48,12 @@ export default function RelatedTask({
   };
 
   return (
-    <Container>
+    <div
+      onClick={() => {
+        if (onClick != null) onClick(task.id);
+      }}
```
Comment (tgargula) [inline cabra/src/pages/components/RelatedTask.tsx:54]:
```suggestion
      onClick={() => onClick?.(task.id)}
```
Looks better, doesn't it? 😄

File: cabra/src/pages/components/RelatedTaskList.tsx [hunk 2]
```diff
@@ -1,46 +1,51 @@
...
-    <Container>
+    <div className={className}>
       {subtasks.map((relatedTask) => (
         <RelatedTask
           key={relatedTask.relationId}
           task={relatedTask.task}
           relationType={relationType}
           parentTaskId={parentTaskId}
+          onClick={(taskId: string) => {
+            navigate("../task/" + taskId);
```
Comment (tgargula) [inline cabra/src/pages/components/RelatedTaskList.tsx:45]: you should use a route helper

File: cabra/src/pages/components/TaskDetailsCard.tsx [hunk 3]
```diff
@@ -70,7 +69,11 @@ export default function TaskDetails({ id }: Props) {
           {task.description || "<No description>"}
         </CardField>
         <CardField title="Subtasks" span={3}>
-          <SubtasksList parentTaskId={id} />
+          <RelatedTasksList
+            relationType={DirectedRelationType.Subtask}
+            parentTaskId={id}
+            tw="flex"
```
Comment (tgargula) [inline cabra/src/pages/components/TaskDetailsCard.tsx:75]: I think you can move this 'flex' to the component

File: cabra/src/pages/components/RelatedTaskList.tsx [hunk 4]
```diff
@@ -1,46 +1,51 @@
...
+    return <div>{"<No tasks>"}</div>;
   }
 
   return (
-    <Container>
+    <div className={className}>
       {subtasks.map((relatedTask) => (
         <RelatedTask
           key={relatedTask.relationId}
           task={relatedTask.task}
```
Comment (tgargula) [inline cabra/src/pages/components/RelatedTaskList.tsx:41]: According to the most popular Airbnb style guide, we should use destructing here
```suggestion
      {subtasks.map(({ relationId, task }) => (
        <RelatedTask
          key={relatedTask.relationId}
          task={relatedTask.task}
```
but the change is optional

**Review by tgargula — COMMENTED** [review]
Nicely done! Take a look at the comments

File: cabra/src/pages/components/TaskDetailsCard.tsx (hunk 3 above)
Comment (koszar91) [reply cabra/src/pages/components/TaskDetailsCard.tsx:75]: Not really. The RelatedTaskList component is used in two ways: horizontally (subtasks) and vertically (other relations). I consider it better to generify it this way.

**Comment by koszar91** [comment]
TODO: prevent event bubbling when clicking checkobx on the subtask

**Comment by gregori0o** [comment]
I think that button `Go Back` in details card should change view to tasks list. If we click a few times related tasks, It will be hard to return to main view.

**Review by tgargula — APPROVED** [review]
😎

**Pull request merged**